.github/workflows/ci_build_merge_manual.yml

```diff
@@ -29,17 +29,28 @@ jobs:
         notebook_files=$(find notebooks -name "*.ipynb" | jq -R -s -c 'split("\n")[:-1]')
         echo "notebook_list=$notebook_files" >>$GITHUB_OUTPUT
 
+    ## Determine if we need a large sized runner from the config file
+    - name: Parse CI Config
+      id: parse-config
+      run: |
+        # Convert ci_config.txt to JSON with group structure for specific runners, string for default
+        config=$(cat ci_config.txt | jq -R 'split(":") | { (.[0]): { "group": .[1] } }' | jq -s 'reduce .[] as $item ({}; . + $item) + {"default": "ubuntu-latest"}' | jq -c .)
```

Comment:
Let's swap this to run on Ubuntu 24.04 for the actual CI portions of the run. This will ensure we're consistent across the board between the free runners' OS and the static large runner's OS as we cannot set the static runners to latest.